### ISSUE TITLE
fix denoising artifacts

### ISSUE THREAD

**Issue by georgeslabreche** [georgeslabreche/opssat-onboard-image-denoiser#53]
## Description
1. Sometimes, denoising produces "colorful" pixels instead of the expected white/near-white ones.
2. It this a limitation from the model or a bug in the denoiser when [the image is denormalized](https://github.com/georgeslabreche/opssat-onboard-image-denoiser/blob/ec95812af0bf8a0a4c3d4c1f24d049eaf23c4c3a/tools/denoiser/src/denoiser.cpp#L622)?
3. Attached are examples from the spacecraft with WGANs denoising for FPN-50. Patch sizes are 56x56 with a 6 pixel margin (refer to logs for exact denoising commands): [onboard-denoising-examples.zip](https://github.com/georgeslabreche/opssat-onboard-image-denoiser/files/12915372/onboard-denoising-examples.zip)

## How to investigate
Inspect the normalized and denormalized RGB values for some of the _problematic_ RGB pixels:
1. Compile the denoiser to run it locally and inspect the denoised RGB pixel values before and after denormalization. If the denormalized values are not what is expected given their normalized values then it's a problem with denormalization.
2. Compare the normalized RGB pixel values of the original image with the normalized RGB pixel of the denoised image. If they are not similar then it's a problem with the model.

## Example 1
|![img_msec_1697282170304_2_thumbnail original resized](https://github.com/georgeslabreche/opssat-onboard-image-denoiser/assets/1258236/1d3353cb-ec2c-4d71-a51e-16ff9fca8365)|![img_msec_1697282170304_2_thumbnail original noised](https://github.com/georgeslabreche/opssat-onboard-image-denoiser/assets/1258236/3c57f84a-c380-4e9e-bacb-3cd2c4e0da89)|![img_msec_1697282170304_2_thumbnail](https://github.com/georgeslabreche/opssat-onboard-image-denoiser/assets/1258236/d4a12965-8da0-4afb-8372-4afe9116dc03)|
|--------------|------------|--------------|
| (a) Original | (b) Noised | (c) Denoised |

## Example 2
|![img_msec_1697282677102_2_thumbnail original resized](https://github.com/georgeslabreche/opssat-onboard-image-denoiser/assets/1258236/9a93fca9-a58c-46a4-b0f2-7485bf93b292)| ![img_msec_1697282677102_2_thumbnail original noised](https://github.com/georgeslabreche/opssat-onboard-image-denoiser/assets/1258236/c48831da-689c-477b-b166-9f89520ba0ae)|![img_msec_1697282677102_2_thumbnail](https://github.com/georgeslabreche/opssat-onboard-image-denoiser/assets/1258236/578d7d52-8ebc-446b-a5ca-18fa195c0205)|
|--------------|------------|--------------|
| (a) Original | (a) Noised | (c) Denoised |

**Comment by cguz** [comment]
Maybe one easy way to test it, is with the option 2. But, I am not sure wich loss function to use.

**Comment by cguz** [comment]
Check it in the last commit in https://github.com/georgeslabreche/opssat-onboard-image-denoiser/pull/57/commits/d346d9b7e16909418a1485065d63e8b5f1a080cb

**Comment by georgeslabreche** [comment]
I wrote the following in the paper:

> The occurrence of false color artifacts in areas with white cloud pixels is likely a result of limited representation of such scenarios in the training dataset. Additionally, the model's potential overfitting to data with sparse cloud coverage could hinder its ability to accurately process and generalize to real-world images with more prevalent cloudiness.
